frontend/app/core/auth/services/mainApiService.js

```diff
@@ -0,0 +1,72 @@
+(function() {
+    angular.module("em")
+    .constant("baseUrl", "http://localhost:2403/")
+    .service("em.mainApiService", mainApiService);
+    function mainApiService($http, baseUrl) {
+
+        this.getItems = function(itemsName) {
+            return $http.get(baseUrl + itemsName + '/').then(
```

Comment:
Your method should return promise. You don't need to keep success or error logic here